### PR TITLE
Add support for Aliyun cloud provider

### PR DESCRIPTION
**Pull request by lu-zhenyu**
With this initial version, one can create or delete virtual machines in the Aliyun. Please consider to merge.

### CODE REVIEW

**Comment by hardikdr** [comment]
@lu-zhenyu  Thanks for the PR. We are currently attending Kubecon and little involved in it. We will review PR soon.

**Comment by hardikdr** [comment]
@lu-zhenyu  Can you please provide your email-id here or maybe e-mail me at hardik.dodiya@sap.com? 
I wanted to discuss the means of testing the PR for us.

**Comment by lu-zhenyu** [comment]
My email is eric.lu@sap.com

**Comment by hardikdr** [comment]
Thanks !!

File: pkg/controller/aliyunmachineclass.go [hunk 1]
```diff
@@ -0,0 +1,181 @@
+package controller
```
[the remark this answers is not in the document]
Comment (lu-zhenyu) [reply pkg/controller/aliyunmachineclass.go]: fix in 59fc0c4

File: pkg/driver/driver_aliyun.go [hunk 2]
```diff
@@ -0,0 +1,270 @@
+package driver
```
[the remark this answers is not in the document]
Comment (lu-zhenyu) [reply pkg/driver/driver_aliyun.go]: fix in 59fc0c4

File: Gopkg.toml [hunk 3]
```diff
@@ -69,3 +69,7 @@ required = [
     name = "k8s.io/code-generator"
     unused-packages = false
     non-go = false
+
+[[constraint]]
+  name = "github.com/aliyun/alibaba-cloud-sdk-go"
```
[the remark this answers is not in the document]
Comment (lu-zhenyu) [reply Gopkg.toml]: fix in 59fc0c4

**Comment by vlerenc** [comment]
Ah, OK. I thought Aliyun it's what it's called in your country Eric? Out of curiosity, why are their two names? You as well seem to have two names/identities?

**Comment by lu-zhenyu** [comment]
New PR has been created at https://github.com/gardener/machine-controller-manager/pull/136.

"yun" means "cloud" in Chinese. So, either "aliyun" or "alicloud" is fine.

However, when coding in "gardener" and "terraformer", we found we have to use "alicloud" due to Terraform's [alicloud provider](https://www.terraform.io/docs/providers/alicloud/index.html). For consistency, we decided to convert "aliyun" to "alicloud".

**Pull request closed without merging**